var/spack/repos/builtin/packages/tcsh/package.py

```diff
@@ -15,9 +15,11 @@ class Tcsh(AutotoolsPackage):
     syntax."""
 
     homepage = "http://www.tcsh.org/"
-    url      = "ftp://ftp.astron.com/pub/tcsh/tcsh-6.20.00.tar.gz"
+    url      = "http://ftp.funet.fi/pub/mirrors/ftp.astron.com/pub/tcsh/tcsh-6.21.00.tar.gz"
 
-    version('6.20.00', '59d40ef40a68e790d95e182069431834')
+    version('6.21.00', sha256='c438325448371f59b12a4c93bfd3f6982e6f79f8c5aef4bc83aac8f62766e972')
```

Comment:
Add the `url=` designator here to specify the non-old URL.  This way, we only need one version-specific `url=` designator, rather than one on every old version (of which there will be many someday).

Reply:
Please check above my comments.